tdwii_plus_examples/tests/test_upspushncreatescu_integration.py

```diff
@@ -73,11 +73,45 @@ def test_run_and_check_log(
         # Ensure the correct Python interpreter is used for the subprocess call
         python_executable = sys.executable
 
+        def read_stdout(process, logger, stdout_lines, server_started_flag):
```

Comment:
This is great.  I could see this getting promoted in to a common test function, especially if we have a convention of having any SCP emitting "SCP server started successfully" once it has finished its startup.  Or pass in the "success" string.